component/monitoring.jsonnet

```
@@ -31,6 +31,26 @@ local service_monitor = com.namespaced(params.namespace, {
   },
 });
 
+local asciiTitleCase(str) = std.asciiUpper(str[0]) + str[1:];
+
+local render_failed_job_alert(type) =
+  local alertconfig = params.job_failed_alerts_for[type];
+  if alertconfig.enabled then
+    params.job_failed_alert_template + alertconfig.overrides + {
+      alert: super.alert % { type: asciiTitleCase(type) },
+      expr: super.expr % { type: type },
+    }
+  else
+    null;
```

Might be worth to only process the alert rule types that we officially support, something like:

```suggestion
local failed_job_types = ['archive', 'backup', 'check', 'prune', 'restore'];
local render_failed_job_alert(type) =
  local alertconfig = params.job_failed_alerts_for[type];
  if std.member(failed_job_types, type) && alertconfig.enabled then
    params.job_failed_alert_template + alertconfig.overrides + {
      alert: super.alert % { type: asciiTitleCase(type) },
      expr: super.expr % { type: type },
    }
  else
    null;
```

Or go all the way and throw an error for unknown failed_job types

```suggestion
local failed_job_types = ['archive', 'backup', 'check', 'prune', 'restore'];
local render_failed_job_alert(type) =
  assert std.member(failed_job_types, type) : 'Unknown failed job type "%s"' % type;
  local alertconfig = params.job_failed_alerts_for[type];
  if alertconfig.enabled then
    params.job_failed_alert_template + alertconfig.overrides + {
      alert: super.alert % { type: asciiTitleCase(type) },
      expr: super.expr % { type: type },
    }
  else
    null;
```

---

> Or go all the way and throw an error for unknown failed_job types

i prefer it this way!